### ISSUE TITLE
Sagas are injected twice!?

### ISSUE THREAD

**Issue by leeseean** [redux-saga/redux-saga#1549]
here is the code,ajaxMainfestJson run twice.
saga.js
````import { call, put, takeEvery/* , takeLatest */ } from 'redux-saga/effects';
import Api from '../api';
import {
    AJAX_REQUESTED
} from './types';
import {
    ajaxSucceeded,
    ajaxFailed,
} from './actions';

// worker Saga: will be fired on AJAX_REQUESTED actions
function* ajaxMainfestJson(action) {
   try {
      const res = yield call(Api.ajaxMainfestJson, action.data);
      yield put(ajaxSucceeded(res.data));
   } catch (e) {
      yield put(ajaxFailed(e.message));
   }
}

/*
  Starts ajaxMainfestJson on each dispatched `AJAX_REQUESTED` action.
  Allows concurrent fetches of user.
*/
function* mySaga() {
  yield takeEvery(AJAX_REQUESTED, ajaxMainfestJson);
}

/*
  Alternatively you may use takeLatest.

  Does not allow concurrent fetches of user. If "AJAX_REQUESTED" gets
  dispatched while a fetch is already pending, that pending fetch is cancelled
  and only the latest one will be run.
*/
/* function* mySaga() {
  yield takeLatest("AJAX_REQUESTED", ajaxMainfestJson);
} */

export default mySaga;
````
store.js

````
import {
    createStore,
    applyMiddleware
} from 'redux';
import createSagaMiddleware from 'redux-saga';
import todoReducer from './reducers/todoReducer';
import rootSaga from './saga';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
    todoReducer,
    applyMiddleware(sagaMiddleware)
);

sagaMiddleware.run(rootSaga);

export default store;````

**Comment by shinima** [comment]
I do not quite understand the problem.

It is totally fine that `ajaxMainfestJson` runs twice. Since you yield `takeEvery(AJAX_REQUESTED, ajaxMainfestJson)` in the `mySaga` so that if you dispatch two AJAX_REQUESTED then `ajaxMainfestJson` will run twice.

**Comment by shinima** [comment]
That's really strange, but I cannot tell why this happens from you code. Could you provide a running environment for me to investigate this?  A [sandbox](https://codesandbox.io/s/m79rn28z9y) would be great.

**Comment by leeseean** [comment]
@shinima  https://github.com/leeseean/react-redux-todolist.git

**Comment by shinima** [comment]
Resolved in leeseean/react-redux-todolist#1.  This is an error from the user code.
